### PR TITLE
Feat/summary report

### PR DESCRIPTION
**Pull request by sifelix**
For flood:

image

For Drought:

image

For Incident:

image

-------
Potential Next steps:

- Add all the columns to the summary report based on Roth's Excel (if too many split the table in two tables)
- Add a row with total if still needed
- Hide empty lines ?
- Hide empty columns ?

### CODE REVIEW

File: apps/frontend/translations/en.json [hunk 1]
```diff
@@ -1,4 +1,8 @@
 {
+  "common": {
+    "province": "Province",
```
Comment (ericboucher) [inline apps/frontend/translations/en.json:3]: these already exist, no?

**Comment by ericboucher** [comment]
Looks like a good start, a few comments:

- note that the text version of province/district/commune should probably not be verticalized, otherwise we don't have enough space to see what's written.
- make the toggles actually switches (should be blue when on, white when off)

File: apps/frontend/components/Report/FloodReport/FloodReport.tsx [hunk 2]
```diff
@@ -25,12 +30,34 @@ export const FloodReport = ({
...
-        <DetailedFloodReport report={report} />
+        <DetailedReportDisasterTable
+          columns={SummaryFloodReportColumnSettings.columns}
+          columnGroup={SummaryFloodReportColumnSettings.columnGroup}
+          data={report}
+          rotateHeader={true}
+        />
       ) : (
-        <BriefFloodReport report={report} />
+        <BriefReportDisasterTable
```
Comment (ericboucher) [inline apps/frontend/components/Report/FloodReport/FloodReport.tsx:55]: Let's rename these maybe?

Comment (ericboucher) [reply apps/frontend/components/Report/FloodReport/FloodReport.tsx]: No, I meant `ProvinceLevelReport`, or something like that. But no biggy

File: apps/frontend/translations/en.json (hunk 1 above)
Comment (ericboucher) [reply apps/frontend/translations/en.json:3]: no it's ok for now

File: apps/frontend/components/DisasterTable/DisasterTable.tsx [hunk 3]
```diff
@@ -38,6 +40,13 @@ export const DisasterTable = ({
           whiteSpace: 'normal !important',
           wordWrap: 'break-word !important',
           lineHeight: 'normal',
+          ...(rotateHeader && {
+            transform: 'rotate(-90deg)',
+            minWidth: '180px',
+          }),
+        },
+        '& .MuiDataGrid-columnHeader': {
+          backgroundColor: '#f5f8ff',
```
Comment (ericboucher) [inline apps/frontend/components/DisasterTable/DisasterTable.tsx:49]: There is no API for those?

**Review by ericboucher — DISMISSED** [review]

**Review by ericboucher — DISMISSED** [review]

**Comment by ericboucher** [comment]
@sifelix any quick way to fix the printing for summary views?

**Comment by wadhwamatic** [comment]
Great work @sifelix! This is a massive improvement. A few inputs from my side:
1. Can we make column widths draggable so users can adjust them as needed? (nice to have)
2. It's odd for me to see the column header text shift when I hover over them
3. I'm hoping we can find a better UI to access the 'manage columns' feature as it's not intuitive and I assume the only reason the hover thing mentioned above is happening. What if we added a kebab menu or ellipses next to the summary / all columns toggle?

**Comment by ericboucher** [comment]
I looked into this a bit and here is what I found, with the library we are currently using:
1. resizing is a Pro feature unfortunately ($15/dev/month)
2. this should be fixable with CSS. @sifelix maybe we could hide the 3-dots and sort arrows by default. I could not find a way to do this through the API but maybe we can do it with some CSS magic.
3. We can create a button to open the menu for "province" for example, and give easier access to "manage columns", see below:
Screenshot 2023-06-02 at 1 16 19 PM

**Pull request merged**